Donor/Repositories/DonorRepository.cs

```diff
@@ -83,11 +82,65 @@ public async Task<IEnumerable<Entities.Donor>> GetAllDonorsAsync()
 
         public async Task<Entities.Donor> GetDonorByIdAsync(int id)
         {
-            return await _context.Donors.FindAsync(id);
+            return await _context.Donors
+                .Include(d => d.Organs)
+                .FirstOrDefaultAsync(d => d.Id == id);
         }
 
 
 
+
+        public async Task UpdateDonorAsync(int donorId, Entities.Donor donorUpdate)
+        {
+            using var transaction = await _context.Database.BeginTransactionAsync();
+            try
+            {
+                // Retrieve the existing donor from the database, including their organs
+                var existingDonor = await _context.Donors
+                    .Include(d => d.Organs)
+                    .FirstOrDefaultAsync(d => d.Id == donorId);
+
+                if (existingDonor == null)
+                {
+                    throw new ArgumentException("Donor not found with the provided ID");
+                }
+
+                existingDonor.FirstName = donorUpdate.FirstName;
```

Comment:
The mapping  should be handle in the profile class (but is just something to think about)